.github/workflows/build_documentation.yml

```diff
@@ -40,14 +43,33 @@ jobs:
           name: nix-community
           authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
 
-      - run: |
+      - name: Build docs
+        run: |
           set -ex
-          mkdir -p docs-build
-          nix build github:nix-community/nixvim#docs
-          cp -r result/share/doc/nixvim/* docs-build
-          nix build github:nix-community/nixvim/nixos-24.05#docs
-          # TODO: use the new path when 24.11 hits
-          cp -r result/share/doc docs-build/stable
+
+          # 1: branch
+          # 2: dest-dir (relative to /nixvim/)
+          build() {
+            flake="github:${repo}${1:+/$1}"
+            baseHref="/${repoName}/${2:+$2/}"
+            destDir="${out}${baseHref}"
+
+            # Build docs for the given flake ref, overriding baseHref in the derivation args
+            nix build --impure --expr '(builtins.getFlake "'"$flake"'").outputs.packages.${builtins.currentSystem}.docs.override {
+              baseHref = "'"$baseHref"'";
+            }'
+
+            # Copy the result to the dest-dir
+            mkdir -p "$destDir"
+            cp -r result/share/doc/* "$destDir"
+          }
+
+          # Install main-branch docs at the top-level
+          build 'main' ''
+
+          # Install 24.05 docs under 'stable'
+          # TODO: consider having `<release>` instead of `stable`
+          build 'nixos-24.05' 'stable'
```

Comment:
To better scale during transition periods where some users may still need docs for the previous stable version, I think it'd be good to have `/24.05/` and `/24.11/` in the URL, instead of `/stable/`.

One way we could provide some backward compatibility wound be to create an HTML meta-redirect file for each HTML file in the stable docs;

```bash
cd $destDir/24.05
mkdir ../stable
for file in **/*.html; do
  echo "TODO HTML template for $file" > ../stable/$file
done
# TODO maybe have special handling for ../stable/404.html
```

(unfortunately we need meta-redirect files because gh-pages doesn't support proper HTTP redirects)

Obviously this isn't in scope for this PR, but I hope it offers a potential direction we can aim for?

Reply:
Yeah I agree that stable URLs would be better, we could avoid breaking them whenever possible